### PR TITLE
Custom Scheduled tasks : adding reboot/shutdown/suspend support

### PR DESCRIPTION
**Pull request by MFlyer**
Refs to #735 - #1306 - #1036

This is a "staging area" for #1036 too : generic "custom scheduled tasks" will be on a different PR

To @schakrava : not ready to be merged, relinting required plus some user alerts and WebUI shutdown alerts

Opening PR to let contributors check/test it :)

Mirko

### CODE REVIEW

**Comment by MFlyer** [comment]
PEP8 and linting done, working on other tasks (ex. rsync) + user warnings

Mirko

**Comment by MFlyer** [comment]
User warning for suspend/shutdown on f8efd58

**Comment by MFlyer** [comment]
To @schakrava : 

Added user alerts on WebUI for scheduled shutdown/reboot, I thing we can review/merge this (to start collecting users feedbacks) then go on with another PR for custom tasks and other "default" tasks (ex. rsync for shares)

Fixes #735 #1306
Mirko

**Comment by MFlyer** [comment]
Fixed conflicts after rebase

Mirko

**Comment by schakrava** [comment]
Nice work @MFlyer . I'll probably review this one after 3.9.0 release, FYI.

**Comment by MFlyer** [comment]
Hi @schakrava, happy to read about 3.9.0, so we're moving to next major?? :)
I was trying to have some major features (ex. custom tasks) on it

**Comment by schakrava** [comment]
Sounds great @MFlyer . I think @phillxnet is also brewing LUKS stuff which is pretty major. I'll do my part to get your changes merged and whatever is necessary. Excited!!!

File: src/rockstor/smart_manager/views/task_scheduler.py [hunk 1]
```diff
@@ -31,14 +31,18 @@
...
+    valid_tasks = ('snapshot', 'scrub', 'reboot',
+                   'shutdown', 'suspend', 'custom')
 
     @staticmethod
     def _validate_input(request):
         meta = {}
         crontab = request.data.get('crontab')
         crontabwindow = request.data.get('crontabwindow')
         meta = request.data.get('meta', {})
+        if 'rtc_hour' in meta:
```
Comment (schakrava) [inline src/rockstor/smart_manager/views/task_scheduler.py]: This if block should go below, after we ensure meta is a dictionary.

File: src/rockstor/storageadmin/static/storageadmin/js/views/add_scheduled_task.js [hunk 2]
```diff
@@ -73,22 +82,51 @@ AddScheduledTaskView = RockstorLayoutView.extend({
...
+        // Hacking Handlebars registerPartial normal usage
+        // Handlebars has registerPartial to help with nested templates
+        // inside another template (ex. Person template having sub-templates
+        // Job, Hobbies, etc). Handlebars.registerPartial(partial, value) with
+        // typeOf(value) == string, while we use value for strings and bools.
+        // Previous version of task page had a "master" part plus
+        // optional-fields div filled only for new tasks, relaying on a messy
+        // general part (if new task -> normal, else render optional-fields)
+        // Now we go with full optional-fields use, serving both when adding
+        // new tasks and when editing existing tasks
```
Comment (schakrava) [inline src/rockstor/storageadmin/static/storageadmin/js/views/add_scheduled_task.js]: Cool!

**Comment by schakrava** [comment]
@MFlyer I am very psyched about this pr! Great job, works perfectly! Take a look at my review comments, and you can address them in future prs. I am merging this without any more delay!!!

**Comment by MFlyer** [comment]
Thanks @schakrava

Question of the day: did you fix those lines of code? If not, going to do it on next rounds :)

**Comment by phillxnet** [comment]
@MFlyer Since I was 'passing by' (in pr #1760) I've taken the liberty to apply one of @schakrava suggestions made by way of a comment in this pr. Hope that is OK. Well done on this one by the way and thanks for the explanatory 'in code' comments; helped me sort a tricky aspect of that pr.

**Pull request merged**